tools/AutoTuner/README.md

```diff
@@ -0,0 +1,205 @@
+         --enable-autoscaling \
+         --min-nodes "1" \
+         --max-nodes "625"
+```
+
+## Create NFS and setup mount options
+
+NOTE: This tutorial requires a working NFS server.
+
+To access the NFS mount point in the Ray cluster we use a NFS helm chart. You
+need to modify the helm chart with the information about your server IP
+and mount point path.
+
+### Install NFS Kubernetes provisioner
+
+```bash
+helm repo add nfs-subdir-external-provisioner https://kubernetes-sigs.github.io/nfs-subdir-external-provisioner
+helm repo update
+```
+
+Before installing the helm chart, double check the fields marked with
+`TODO` on the file `./nfs/nfs-helm-values.yaml`. The `server` and `path`
+need to match the NFS server IP and the path to the exposed folder.
+The `nodeSelector` must match the non-preemptive Kubernetes pool.
+
+```bash
+helm install nfs nfs-subdir-external-provisioner/nfs-subdir-external-provisioner -f nfs/nfs-helm-values.yaml
+```
+
+### Create a PVC
+
+Create the PVC:
+
+```bash
+kubectl create -f nfs/nfs-pvc.yaml
+```
+
+## Ray cluster
+
+The two main files for the helm chart are:
+
+```bash
+helm-chart/templates/raycluster.yaml
+helm-chart/values.yaml
+```
+
+To restrict to a specific node pool, replace POOL_NAME with the name
+of the pool you created on the previous step. For this tutorial we use
+`worker-pool`.
+
+```yaml
+nodeSelector:
+  cloud.google.com/gke-nodepool: POOL_NAME
+```
+
+### Deploy Ray cluster inside Kubernetes
+
+Use Helm to deploy a Ray cluster.
+
+```bash
+helm install autotuner ./helm-chart
+```
+
+### Upgrade deployment
+
+After the initial deployment, if you change the values inside `./helm-chart`
+you will need to upgrade the configuration stored at the Kubernetes cluster.
+
+```bash
+helm upgrade autotuner ./helm-chart
+```
+
+### Useful ways to observe
+
+Ray dashboard:
+
+```bash
+kubectl port-forward service/autotuner-ray-head 8265
+```
+
+Overall cluster status:
+
+```bash
+kubectl get nodes,rayclusters,pods,services,replicaset,pvc,pv
+```
+
+Ray auto-scaler logs:
+
+```bash
+kubectl logs \
+  $(kubectl get pod -l cluster.ray.io/component=operator -o custom-columns=:metadata.name) \
+  --tail=100 -f
+```
+
+### Remove
+
+First, delete the `RayCluster` custom resource.
+
+```bash
+kubectl delete raycluster autotuner
+```
+
+Delete the Ray release.
+
+```bash
+helm uninstall autotuner
+```
+
+## Running Ray programs with Ray Client
+
+### Using port forwarding
+
+NOTE: Ray requires that the version of Python on the server and locally
+match. The current `openroad/ray` image has Python 3.7.7, make sure your
+local machine has at least Python 3.7.x before continuing.
+
+```bash
+pip3 install -U --user 'ray[default,tune]==1.11.0' ax-platform hyperopt nevergrad optuna pandas
+```
+
+Start the port forwarding:
+
+```bash
+kubectl port-forward service/autotuner-ray-head 10001
+```
+
+Run the test script locally:
+
+```bash
+python3 kubernetes/run.py
+```
+
+### Connecting to head node
+
```

Comment:
It seems there are difficult ways for users to configure in Ray. If so, this should be in a separate section and upfront.

Reply:
Yes, thisis one way. I have updated the description of the header with level two (ie., `## Running Ray programs with Ray Client`) to reflect that. All subsections from that section are different ways to start a AutoTuner job.